src/Symfony/Bundle/SecurityBundle/Security/FirewallContext.php

```diff
@@ -25,7 +25,7 @@ class FirewallContext
     private $exceptionListener;
     private $config;
 
-    public function __construct(array $listeners, ExceptionListener $exceptionListener = null, FirewallConfig $config = null)
+    public function __construct(/* iterable */ $listeners, ExceptionListener $exceptionListener = null, FirewallConfig $config = null)
```

Comment:
do we care about lazyness when getListeners is called? I think we do. Right now, it's used only for debugging, where we don't care being lazy or not.